typescriptFixme.ts

```diff
@@ -1,115 +1,143 @@
 // add the type annotation to explicitly specify the type of the variable:
-var isDone = false;
-var lines = 42;
-var name = "Anders";
+var isDone : boolean = false;
+var lines : number = 42;
+var name1 : string = "Anders";
 
 // annotate the type as when it's not known
-let unknownTypeVariable;
+let unknownTypeVariable: unknown;
 
 //annotate this array
-var list = [1, 2, 3];
+var list : number[] = [1, 2, 3];
 
 // implement color in typescript
-var Color = { Red: 0, Green: 1, Blue: 2 };
-var c = Color.Green;
+enum Color {
+  Red = 0, Green = 1, Blue = 2
+  };
+var c : Color = Color.Green;
 console.log(Object.keys(Color)[c]); // "Green"
 
-function littleHorribleAlert() {
+function littleHorribleAlert(): void {
   alert("I am a little annoying box!!");
 }
 
-var f1 = function (i) {
+var f1 = function (i:number): number {
   return i * i;
 };
 // as f1 just don't specify the return type here
-var f2 = function (i) {
+var f2 = function (i:number) {
   return i * i;
 };
 
-var f3 = (i) => {
+const f3 = (i: number): number => {
   return i * i;
 };
 
 //change f4 to accept a string or a number as a parameter
-function f4(i) {
+function f4(i: number|string) {
   console.log("The value was " + i);
 }
 
 // Annotate the person objects with an interface. Some properties are optional
+interface Person {
+  name: string;
+  age?: number;
+  move: () => void;
+}
+
 var person1 = { name: "Bobby", move: function () {} };
-var person2 = { name: "Bobby", age: 42, move: function () {} };
+var person2 = { name: "Bobby", age: 42, move: function () { } };
+
 
 // Annotate mySearch to describe a function that accepts two string parameters
-var mySearch = function (src, sub) {
+var mySearch: (src: string, sub: string) => boolean = function (src, sub) {
   return src.search(sub) != -1;
 };
-
 //annotate the constructor arguments and properties
 class Point {
-  x = 0;
-  constructor(x, y = 0) {
+  x: number = 0;
```

Comment:
no need to assign x and y to zero if you they are assigned in the constructor. Just declare `x` and `y` without initializing them. 
```ts
x: number;
y: number;
```